misk-service/src/test/kotlin/misk/ServiceGraphBuilderTest.kt

```diff
@@ -49,8 +49,8 @@ class ServiceGraphBuilderTest {
     assertThat(script).isEqualTo(
       """
         |starting Service A
-        |starting Enhancement A
         |starting Service C
+        |starting Enhancement A
```

Comment:
Is this not a significant behaviour change? Enhancements of a service are meant to start before dependents of the service.

Reply:
oh interesting, good catch. I think it's fine in this case though given it's just `SchemaMigratorService`

Reply:
Maybe we ought to just rip out service enhancements entirely and add a `.dependsUpon` directive to allow expressing the inverse dep relationship that "enhancements" are now expressing 🤔